src/nanotron/helpers.py

```diff
@@ -91,7 +93,7 @@ def lr_scheduler_builder(optimizer: Optimizer, lr_scheduler_args: LRSchedulerArg
     else:
         lr_decay_starting_step = lr_scheduler_args.lr_decay_starting_step
 
-    def lr_lambda(current_step: int):
+    def lr_lambda(current_step: int, lr: float):
```

Comment:
the lr in the arg is confusing. Before, it was clear that `lr_scheduler_args.learning_rate`is a constant coming from the config

Reply:
because in µTransfer, each parameter has a custom learning rate (custom_lr = lr_scheduler_args.learning_rate * scaling_factor), so in standard paramtrization, indeed lr_lambda only takes a single learning rate lr_scheduler_args.learning_rate * scaling_factor, but in mup, each parameter has a different learning rate

Reply:
can we rename `lr` to something more explanatory like `initial_lr` or `prev_lr`? and add docs for it